osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -260,6 +260,9 @@ public WorkingBeatmap GetWorkingBeatmap(BeatmapInfo beatmapInfo, WorkingBeatmap
             if (beatmapInfo?.BeatmapSet == null || beatmapInfo == DefaultBeatmap?.BeatmapInfo)
                 return DefaultBeatmap;
 
+            var info = beatmapInfo;
+            beatmapInfo = QueryBeatmap(p => info.OnlineBeatmapID == p.OnlineBeatmapID) ?? beatmapInfo;
```

Comment:
what is this

Reply:
This is querying the local database for a local version of the beatmap, the logic seemed to have been missing before.

Reply:
I can't tell what this change was made for, not if it is correct. It also doesn't look directly related to this PR or the issues it's fixing, so please at very least split into its own PR.